contracts/allocators/LUSDAllocator.sol

```diff
@@ -245,34 +302,57 @@ contract LUSDAllocator is Ownable {
 
         // 2.  Stake LQTY rewards from #1.  This txn will also give out any outstanding ETH+LUSD rewards from prior staking
         uint256 balanceLqty = IERC20(lqtyTokenAddress).balanceOf(address(this)); // LQTY balance received from stability pool
-        
-        uint stakingEthRewards = 0;
-        uint stakingLUSDRewards = 0;
         if (balanceLqty > 0) {
-            stakingEthRewards = lqtyStaking.getPendingETHGain(address(this));
-            stakingLUSDRewards = lqtyStaking.getPendingLUSDGain(address(this));
             //Stake
-            IERC20(lqtyTokenAddress).approve(address(lqtyStaking), balanceLqty); // approve to deposit into stability pool
-            lqtyStaking.stake(balanceLqty); //Stake LQTY, also receives any prior ETH+LUSD rewards from prior staking TODO need to deposit this LUSD
+            lqtyStaking.stake(balanceLqty); //Stake LQTY, also receives any prior ETH+LUSD rewards from prior staking
         }
 
-        // 3.  Deposit LUSD from #2 into StabilityPool
-        if (stakingLUSDRewards > 0) {
-            IERC20(lusdTokenAddress).approve(address(lusdStabilityPool), stakingLUSDRewards); // approve to deposit into stability pool
-            lusdStabilityPool.provideToSP(stakingLUSDRewards, frontEndAddress);
+        // 3.  If we have eth, convert to weth, then swap a percentage of it to LUSD.  If swap successul then send all remaining WETH to treasury
+        uint256 ethBalance = address(this).balance;  // Use total balance in case we have leftover from a prior failed attempt
+        if (ethBalance > 0) {
+            // Wrap ETH to WETH
+            weth.deposit{value: ethBalance}();
+
+            uint256 wethBalance = weth.balanceOf(address(this));  //Base off of WETH balance in case we have leftover from a prior failed attempt
+            uint256 amountWethToSwap = wethBalance * percentETHtoLUSD / 100;
+
+            // Approve WETH to uniswap
+            weth.safeApprove(address(swapRouter), amountWethToSwap);
+
+            // Multiple pool swaps are encoded through bytes called a `path`. A path is a sequence of token addresses and poolFees that define the pools used in the swaps.
+            // The format for pool encoding is (tokenIn, fee, tokenOut/tokenIn, fee, tokenOut) where tokenIn/tokenOut parameter is the shared token across the pools.
+            // Since we are swapping WETH to DAI and then DAI to LUSD the path encoding is (WETH, 0.3%, DAI, 0.3%, LUSD).
+            ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+                path: abi.encodePacked(address(weth), poolFee, hopTokenAddress, poolFee, lusdTokenAddress),
+                recipient: address(this),  //Send LUSD here
+                deadline: block.timestamp,
+                amountIn: amountWethToSwap,
+                amountOutMinimum: 0
```

Comment:
Does this zero here open us up to 100% slippage?

Reply:
Nice catch.  I copied the example from https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps

but i missed this part:

```
amountOutMin: the minimum amount of the outbound asset, less than which will cause the transaction to revert. For the sake of this example we will set it to 0, in production one will need to use the SDK to quote an expected price, or an on chain price oracle for more advanced manipulation resistant systems.
```

Reply:
Passed in via parameter now